vault/resource_ldap_auth_backend_test.go

```diff
@@ -284,7 +285,8 @@ resource "vault_ldap_auth_backend" "test" {
     discoverdn             = false
     deny_null_bind         = true
     description            = "example"
-	userfilter             = "({{.UserAttr}}={{.Username}})"
+    userfilter             = "({{.UserAttr}}={{.Username}})"
+    username_as_alias      = true
```

Comment:
I thought it would be checked automatically in the loop further below in function `testLDAPAuthBackendCheck_attrs`?

https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_ldap_auth_backend_test.go#L222

Therefore, I only added it to the list `attrs` and let the loop do its magic to check with the state after the test.

Reply:
I believe that just compares the TF state to Vault's. It doesn't explicitly validate what the test TF provided. Adding  a `resource.TestCheckResourceAttr()` for the field ensures that the test input is honored througout.

Reply:
Unfortunately I don't know the code base good enough to recognize what needs to be changed in the tests here. Would you mind explaining again differently what needs to be done here or take over the actual implementation? Thanks a lot.

Reply:
Ah, I see that the tests for this resource are somewhat non-standard. We can update them outside of this PR.

A more standard test starts off by validating the config to the state e.g.:

https://github.com/hashicorp/terraform-provider-vault/blob/7afeebd07ac81cda40cf19010212146928b6ad54/vault/resource_pki_secret_backend_sign_test.go#L175